Bug Report: MultiplayerTemplate failed on export launcher for windows

**Describe the bug**

==================================================================
[INFO] root:   MTPSModule.cpp

[INFO] root:   MTPS.Server.vcxproj -> C:\Users\RalphSobremonteL\O3DE\Projects\MTPS\build\launcher\lib\release\MTPS.Server.lib

[INFO] root:   Copying MTPS.ServerLauncher runtime dependencies to output...

[INFO] root:   -- Copying "C:/Users/RalphSobremonteL/.o3de/3rdParty/packages/vulkan-validationlayers-1.3.261-rev1-windows/vulkan-validationlayers/lib/release/VkLayer_khronos_validation.json" to "C:/Users/RalphSobremonteL/O3DE/Projects/MTPS/build/launcher/bin/release/"...

[INFO] root:   -- Copying "C:/Users/RalphSobremonteL/.o3de/3rdParty/packages/vulkan-validationlayers-1.3.261-rev1-windows/vulkan-validationlayers/lib/release/VkLayer_khronos_validation.dll" to "C:/Users/RalphSobremonteL/O3DE/Projects/MTPS/build/launcher/bin/release/"...

[INFO] root:   -- Copying "C:/Users/RalphSobremonteL/.o3de/3rdParty/packages/PhysX-4.1.2.29882248-rev8-windows/PhysX/shared//bin/PhysXDevice64.dll" to "C:/Users/RalphSobremonteL/O3DE/Projects/MTPS/build/launcher/bin/release/"...

[INFO] root:   -- Copying "C:/Users/RalphSobremonteL/.o3de/3rdParty/packages/PhysX-4.1.2.29882248-rev8-windows/PhysX/shared//bin/PhysXGpu_64.dll" to "C:/Users/RalphSobremonteL/O3DE/Projects/MTPS/build/launcher/bin/release/"...

[INFO] root:   Building Custom Rule E:/O3DE/CMakeLists.txt

[INFO] root:   unity_0_cxx.cxx

[INFO] root:      Creating library C:/Users/RalphSobremonteL/O3DE/Projects/MTPS/build/launcher/lib/release/MTPS.ServerLauncher.lib and object C:/Users/RalphSobremonteL/O3DE/Projects/MTPS/build/launcher/lib/release/MTPS.ServerLauncher.exp

[INFO] root: unity_0_cxx.obj : error LNK2019: unresolved external symbol CreateModuleClass_Gem_MTPS_Server referenced in function CreateStaticModules [C:\Users\RalphSobremonteL\O3DE\Projects\MTPS\build\launcher\o3de\MTPS.ServerLauncher.vcxproj]

[INFO] root: C:\Users\RalphSobremonteL\O3DE\Projects\MTPS\build\launcher\bin\release\MTPS.ServerLauncher.exe : fatal error LNK1120: 1 unresolved externals [C:\Users\RalphSobremonteL\O3DE\Projects\MTPS\build\launcher\o3de\MTPS.ServerLauncher.vcxproj]

[INFO] root: 

[INFO] root: Terminating process 'cmake' with PID(7084)

[INFO] root: process 'cmake' with PID(7084) terminated with exit code 1

Error building the monolithic launcher(s).
==================================================================

**Steps to reproduce**
Steps to reproduce the behavior:
Export Launcher to Windows

**Expected behavior**
Produce an executable

**Actual behavior**
Failed to produce export launcher

**Found in Branch**
o3de main

**Desktop/Device (please complete the following information):**
Windows 10

--- a/Templates/Multiplayer/Template/Gem/Source/${Name}Module.cpp
+++ b/Templates/Multiplayer/Template/Gem/Source/${Name}Module.cpp
@@ -43,8 +43,24 @@ namespace ${SanitizedCppName}
     };
 }// namespace ${SanitizedCppName}
 
+#if defined(AZ_MONOLITHIC_BUILD)
+    // Monolithic client
+    #if defined(O3DE_GEM_NAME)
+        AZ_DECLARE_MODULE_CLASS(AZ_JOIN(Gem_, O3DE_GEM_NAME, _Client), ${SanitizedCppName}::${SanitizedCppName}Module)
+    #else
+        AZ_DECLARE_MODULE_CLASS(Gem_${SanitizedCppName}_Client, ${SanitizedCppName}::${SanitizedCppName}Module)
+    #endif
+
+    // Monolithic server
+    #if defined(O3DE_GEM_NAME)
+        AZ_DECLARE_MODULE_CLASS(AZ_JOIN(Gem_, O3DE_GEM_NAME, _Server), ${SanitizedCppName}::${SanitizedCppName}Module)
+    #else
+        AZ_DECLARE_MODULE_CLASS(Gem_${SanitizedCppName}_Server, ${SanitizedCppName}::${SanitizedCppName}Module)
+    #endif
+#endif
+
 #if defined(O3DE_GEM_NAME)
-AZ_DECLARE_MODULE_CLASS(AZ_JOIN(Gem_, O3DE_GEM_NAME), ${SanitizedCppName}::${SanitizedCppName}Module)
+    AZ_DECLARE_MODULE_CLASS(AZ_JOIN(Gem_, O3DE_GEM_NAME), ${SanitizedCppName}::${SanitizedCppName}Module)
 #else
-AZ_DECLARE_MODULE_CLASS(Gem_${SanitizedCppName}, ${SanitizedCppName}::${SanitizedCppName}Module)
+    AZ_DECLARE_MODULE_CLASS(Gem_${SanitizedCppName}, ${SanitizedCppName}::${SanitizedCppName}Module)
 #endif

